### PR TITLE
Add x-requested-with to headers proxy

### PR DESCRIPTION
**Pull request by vovakulikov**
Follow up for https://github.com/sourcegraph/sourcegraph/pull/27313

Based on the PR above we should send all requests with a special header `x-requested-with` in order to pass trust logic on BE request resolver.

### CODE REVIEW

File: client/web/dev/utils/csrf/get-csrf-token-cookie-middleware.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { CSRF_COOKIE_NAME } from './get-csrf-token-and-cookie'
 
 // Attach `CSRF_COOKIE_NAME` cookie to every response to avoid "CSRF token is invalid" API error.
 export const getCSRFTokenCookieMiddleware = (csrfCookieValue: string): RequestHandler => (_request, response, next) => {
+    _request.headers['x-requested-with'] = 'Sourcegraph FE local server'
```
Comment (umpox) [inline client/web/dev/utils/csrf/get-csrf-token-cookie-middleware.ts:7]:
```suggestion
export const getCSRFTokenCookieMiddleware = (csrfCookieValue: string): RequestHandler => (request, response, next) => {
    request.headers['x-requested-with'] = 'Sourcegraph FE local server'
```

I'm assuming the variable was previously prefixed with `_` as it was unused

**Review by umpox — APPROVED** [review]
Thank you!

**Review by valerybugakov — APPROVED** [review]

File: client/web/dev/utils/csrf/get-csrf-token-cookie-middleware.ts [hunk 2]
```diff
@@ -3,7 +3,8 @@ import { RequestHandler } from 'express'
 import { CSRF_COOKIE_NAME } from './get-csrf-token-and-cookie'
 
 // Attach `CSRF_COOKIE_NAME` cookie to every response to avoid "CSRF token is invalid" API error.
-export const getCSRFTokenCookieMiddleware = (csrfCookieValue: string): RequestHandler => (_request, response, next) => {
+export const getCSRFTokenCookieMiddleware = (csrfCookieValue: string): RequestHandler => (request, response, next) => {
+    request.headers['x-requested-with'] = 'Sourcegraph FE local server'
```
Comment (valerybugakov) [inline client/web/dev/utils/csrf/get-csrf-token-cookie-middleware.ts:7]: Maybe it makes sense to leave a comment with the PR that we mentioned in the Slack thread for future reference.

Comment (vovakulikov) [reply client/web/dev/utils/csrf/get-csrf-token-cookie-middleware.ts:10]: Good point. I added comment with a link to Stephen PR.

**Pull request merged**